Automattic/wp-calypso
# Packages: Move tree-select to packages

Also update the build to produce commonJS and ESM modules.

Testing Instructions
* `npm run build` should now also run `npm run build-packages`
* after `npm run build-packages`, node_modules/@automattic/tree-select should be a symlink to the local tree-select in packages
* lerna bootstrap should run the build in the tree-select package, which should generate an ESM and CJS build there
* npm start should spin up the dev server and calypso should pull the code for tree-select from the packages directory, specifically from the ESM build
* changing the code in packages/tree-select and rebuilding (either npm run build there or lerna run build from the root of the project) should make the dev server rebuild calypso
* basic lerna commands ( `lerna changed` ) should work

Todo:
 * how do we handle lerna publish with 2fa otp?
 * how / when should we publish and version?
